packages/eslint-config-small-ads/package.json

```diff
@@ -0,0 +1,27 @@
+{
+  "name": "eslint-config-small-ads",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
```

Comment:
I think this is a good opotunity to discuss the difference between `devDependencies` and `dependencies`. Can you tell me whats the difference?

Reply:
The `dependencies` are require for the project to run, and therefore one should try to keep them small. The `devDependencies` are only required to develop the project. They can be big because they don't go in the build

Reply:
The only dependency that I think it was weird to include in the `devDependencies` was `"typescript": "^4.1.3"`. But I got errors if I don't do it.

Reply:
I am watching [this](https://www.youtube.com/watch?v=ZSdCNf1ncOE) conference on lerna now.

Reply:
Super you know what they are. Now think _only_ about the `eslint-config-small-ads` package. And think of it as its own independent app. 

Which are the required dependencies for this project? Or in other words, which dependencies does the `eslint-config-small-ads` has the responsibility to install so that it operates correctly when we are to reuse it in some project?

Reply:
Hmm, well if I focus only on eslint-config-small-ads I would install only the general devDependencies, those that could be reusable in whatever other project. The thing is, we know we want to use Typescript in this project, so I would include also those, and I would leave out the ones related to react.
"devDependencies": { "@typescript-eslint/eslint-plugin": "^4.14.0", "@typescript-eslint/parser": "^4.14.0", "eslint": "^7.18.0", "eslint-config-airbnb": "^18.2.1", "eslint-config-prettier": "^7.2.0", "eslint-import-resolver-typescript": "^2.3.0", "eslint-plugin-import": "^2.22.1", "eslint-plugin-prettier": "^3.3.1", "prettier": "^2.2.1" }

Reply:
use ``` instead of one ` ;)

Reply:
What you say isnt wrong. Also you are right including typescript and react related dependencies here. Perhaps not typescript itself but the eslint typescript/react related dependencies we want. 

Let me rephrase and be more specific now. 
1. We want this eslint config to be used by react projects that use typesript. 
2. Also we want to get straight, which dependencies are `devDependencies` and which are simply `dependencies`.

Lets focus our attention on that second requirement now. You are saying that dependencies are required for the project to run and devDependencies are used only in development. For example in a react project devDependency could be some testing library because when we build the react app and we make it a production executable that runs on the browser the testing library will not be necessary. 

Now I want us to think of the same way about this package. In the future we will install this package in some react project. For this project to run some dependencies are necessary and some dependencies are not necessary. 
Which would be necessary and which not?

Reply:
I will let you think about it :) It is a riddle because I dont wanna give you the answer. I think you will benefit more understanding this by doing your own research

Reply:
Ooooh, I think I go it! Shit, I was installing the packages as devDependencies, and because of that, when I was running `npm i ../packages/eslint-config-small-ads` on the other projects nothing was being installed.

Reply:
But maybe I am wrong, I thought of that while taking a shower :D I didn't read anything new I just thought about it. So maybe I have to keep thinking and researching

Reply:
Unfortunately Maria and I had the plan of watching a movie. Now I am being forced to "relax" I cannot keeping studying, boomer :(.

Reply:
Great great! 

RELAX. I also need to ;) We do that tomorrow

Reply:
Good Morning @paschalidi, I changed the `devDependencies` and made them `dependencies` in the `eslint-config-small-ads` project, but it is still not working. 
I installed `eslint-config-small-ads` in `fe-small-advertisements` but still looks as if I have to install everything manually.

Reply:
Well, there is a reason for that. I neglected to mention, that while you work with lerna, we dont navigate into each project and manually doing npm i for each one of them. What we are doing is basically asking lerna to do this work for us. This way it happens in parallel for all the different projects lerna is responsible for. 

What you want to do is read these commands [here](https://github.com/lerna/lerna). Lerna bootstrap specifically.

Reply:
Oh, I just saw this, let me see